src/models/User.js

```diff
@@ -0,0 +1,62 @@
+const mongoose = require('mongoose');
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcryptjs');
+
+const userSchema = new mongoose.Schema({
+    username: {
+        type: String,
+        required: [true, 'Cannot be blank'],
+        match: [/^[a-zA-Z0-9]+$/, 'Username is invalid'],
+        unique: [true, 'This username is already taken'],
+        maxlength: 50,
+    },
+    firstName: {
+        type: String,
+        required: [true, 'Cannot be blank'],
+        match: [/^[a-zA-Z0-9]+$/, 'Username is invalid'],
+        maxlength: 50,
+    },
+    lastName: {
+        type: String,
+        required: [true, 'Cannot be blank'],
+        match: [/^[a-zA-Z0-9]+$/, 'Username is invalid'],
+        maxlength: 50,
+    },
+    role: {
+        type: String,
+        enum: ['Mentor', 'Student', 'Apprentice', 'Admin'],
+        required: [true, 'Please choose your role']
+    },
+    email: {
+        type: String,
+        required: [true, 'Cannot be blank'],
+        match: [/\S+@\S+\.\S+/, 'is invalid'],
+        unique: [true, 'This email is already taken.'],
+    },
+    password: {
+        type: String,
+        required: [true, 'Cannot be blank'],
+    },
+});
+
+userSchema.pre('save', async function() {
+    const salt = await bcrypt.genSalt(10);
+    this.password = await bcrypt.hash(this.password, salt)
+})
+
+userSchema.methods.createJWT = function () {
```

Comment:
[Structure]
Generally these would be in the user controller not here. This is the model file, so you generally put things here that have to do with the data model.
The creation of the JWT and comparison of the password isn't about the data model itself (how the data is laid out and structured) its about the logic used to perform certain API functions. Specifically a user proving they are who they say they are (Authentication). Logic related to using the API is what the controller is for.

Reply:
You actually have this code in the controller it looks like, you just used it for login, but not for register. You want to use it for both.

Reply:
I moved the functions/methods in the controllers folder in an auth.js file. Please let me know if this is the right approach. 
I also restructured the registration as suggested. I also tested it on Postman, and it works.